### PR TITLE
Add types for empower-core and improve types for empower

### PR DESCRIPTION
**Pull request by jirutka**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/power-assert-js/empower/blob/master/index.js, https://github.com/power-assert-js/empower/blob/master/lib/default-options.js
- ~~[ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~

### CODE REVIEW

File: types/empower-core/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+// Type definitions for empower-core 1.2
+// Project: https://github.com/twada/power-assert-runtime
+// Definitions by: Jakub Jirutka <https://github.com/me>
```
Comment (amcasey) [inline types/empower-core/index.d.ts:3]: Please update the URL as well.

File: types/empower-core/tslint.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-redundant-jsdoc-2": false,
```
Comment (amcasey) [inline types/empower-core/tslint.json:4]: Please suppress the rules on individual lines, rather than globally.

File: types/empower/tslint.json [hunk 3]
```diff
@@ -1,11 +1,7 @@
 {
     "extends": "dtslint/dt.json",
     "rules": {
-        "dt-header": false,
-        "no-var-keyword": false,
-        "prefer-const": false,
-        "strict-export-declare-modifiers": false,
-        "typedef-whitespace": false,
-        "whitespace": false
+        "no-redundant-jsdoc-2": false,
```
Comment (amcasey) [inline types/empower/tslint.json:4]: As above

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/empower/tslint.json (hunk 3 above)
Comment (jirutka) [reply types/empower/tslint.json:4]: I don’t understand this rule, it seems contraproductive to me. I’m using `@param` and `@return` annotations for documentation, not types declaration. JSdoc documentation in types is very helpful and many type declarations contain it. So why?

Comment (amcasey) [reply types/empower/tslint.json:4]: I don't believe there's any restriction on using JSDoc without types.  Are you getting errors when you re-enable the lint rule?  I removed the rules block locally (from both projects) and didn't see any errors.

File: types/empower-core/tslint.json (hunk 2 above)
Comment (jirutka) [reply types/empower-core/tslint.json]: Done

File: types/empower-core/index.d.ts (hunk 1 above)
Comment (jirutka) [reply types/empower-core/index.d.ts:3]: Done

File: types/empower/tslint.json (hunk 3 above)
Comment (jirutka) [reply types/empower/tslint.json]: I’d swear that it failed on CI before, but it’s clear now.

File: types/empower-core/index.d.ts (hunk 1 above)
Comment (amcasey) [reply types/empower-core/index.d.ts:3]: Looks like it still says `https://github.com/me`, but should say `https://github.com/jirutka`.

**Review by amcasey — CHANGES_REQUESTED** [review]
Sorry, one more thing

File: types/empower-core/index.d.ts (hunk 1 above)
Comment (jirutka) [reply types/empower-core/index.d.ts:3]: Oh, yes, I’m blind. :facepalm:

**Review by amcasey — APPROVED** [review]

**Comment by jirutka** [comment]
Ready to merge

**Pull request merged**